src/Compilers/CSharp/Portable/CommandLine/CommandLineDiagnosticFormatter.cs

```diff
@@ -66,7 +66,9 @@ internal string RelativizeNormalizedPath(string normalizedPath)
             var directory = PathUtilities.GetDirectoryName(normalizedPath);
             if (string.Compare(directory, 0, normalizedBaseDirectory, 0, normalizedBaseDirectory.Length, StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
It looks like this check does not handle `baseDirectory: "C:\AB"` and `normalizedPath: "C:\ABCD\file.cs"`. Should it?